lms/djangoapps/ccx/api/v0/tests/test_views.py

```diff
@@ -696,6 +752,20 @@ def test_patch_detail_coach_does_not_exist(self):
         resp = self.client.patch(self.detail_url, data, format='json', HTTP_AUTHORIZATION=self.auth)
         self.expect_error(status.HTTP_404_NOT_FOUND, 'coach_user_does_not_exist', resp)
 
+    def test_patch_detail_wrong_modules(self):
+        """
+        Specific test for the case when the input data is valid but the
+        course modules do not belong to the master course
+        """
+        data = {
+            'course_modules': [
+                'block-v1:org.0+course_0+Run_0+type@chapter+block@chapter_foo',
+                'block-v1:org.0+course_0+Run_0+type@chapter+block@chapter_bar'
+            ]
+        }
+        resp = self.client.patch(self.detail_url, data, format='json', HTTP_AUTHORIZATION=self.auth)
+        self.expect_error(status.HTTP_400_BAD_REQUEST, 'course_module_list_not_belonging_to_master_course', resp)
+
```

Comment:
Nit: It'd be nice to have a test that mixed both valid and invalid `course_modules` in the same POST and PATCH call to assert that it rejected all those modules.